opencog/atoms/core/Variables.cc

```diff
@@ -283,7 +293,12 @@ bool VarScraper::is_variable(const Handle& h)
 
 bool VarScraper::is_ordered_link(const Handle& h)
 {
-	return nameserver().isA(h->get_type(), ORDERED_LINK);
+	return is_ordered_type(h->get_type());
+}
+
+bool VarScraper::is_ordered_type(Type t)
+{
+	return nameserver().isA(t, ORDERED_LINK);
```

Comment:
This is guaranteed to lead to bugs.  A better test would be to look for "not unordered" links.   There are rare cases where a user can say just "Link", and just-plain Link.s are ... ordered; whereas the above code assumes the opposite.  (Yes, I was already bitten by this in the pattern matcher)

Reply:
Thanks!